### ISSUE TITLE
Different response on Nighthawk R7800 on latest version

### ISSUE THREAD

**Issue by hartmannaf** [MatMaul/pynetgear#22]
Router: Nighthawk R7800
Firmware: V1.0.2.36

After one of the lastest updates the response had changed:

```
<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope
   xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/"
   SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
<SOAP-ENV:Body>
<m:GetAttachDeviceResponse xmlns:m="urn:NETGEAR-ROUTER:service:DeviceInfo:1">    <NewAttachDevice>
@1;192.168.***.***;HOME-SERVER;18:66:**:**:**:**;wired;;100;Allow@2;192.168.***.***;&lt;unknown&gt;;B0:B9:**:**:**:**;wired;;100;Allow@3;192.168.***.***;HARTIES-WP;48:86:**:**:**:**;wireless;38;52;Allow@4;192.168.***.***;HARTIES-PC;1C:1B:**:**:**:**;wired;;100;Allow@5;192.168.***.***;HUAWEI_P9_LITE;A4:71:**:**:**:**;wireless;56;64;Allow@6;192.168.***.***;KIDS-PI;74:DA:**:**:**:**;wireless;70;62;Allow@7;192.168.***.***;HP08B22A;84:34:**:**:**:**;wireless;56;86;Allow@8;192.168.***.***;*****-HANDY;38:F2:**:**:**:**;wireless;146;68;Allow@9;192.168.***.***;****PI;B8:27:**:**:**:**;wired;;100;Allow@10;192.168.***.***;WINDOWS-PHONE;9C:6C:**:**:**:**;wireless;70;70;Allow@11;192.168.***.***;ANDROID-*****;0C:4E:**:**:**:**;wireless;25;48;Allow@12;192.168.***.***;KELLER_*****_WALL-****;60:01:**:**:**:**;wireless;70;80;Allow@13;192.168.***.***;KELLER_MAIN-****;60:01:**:**:**:**;wireless;70;72;Allow@14;192.168.***.***;SONOFF-****;60:01:**:**:**:**;wireless;70;56;Allow</NewAttachDevice>
</m:GetAttachDeviceResponse>
<ResponseCode>000</ResponseCode>
</SOAP-ENV:Body>
</SOAP-ENV:Envelope>
```

To get pynetgear working again with home assistent and my router I needed to add the re.DOTALL param to the regex in `get_attached_devices`
I also needed to allow parsing if the length is 8 in `get_attached_devices`
The router also does not send the number of devices anymore.

**Comment by OneGneissGuy** [comment]
@hartmannaf 
I think I'm having the same problem after updating router firmware.
I'd like to edit get_attached_devices and see what happens. 
How did you get access to and modify the pynetgear scripts on your hassio instance?

**Comment by hartmannaf** [comment]
I cloned the repository and installed pynetgear manually inside of the virt enviroment, after that it automaticly used my modified version

**Comment by MatMaul** [comment]
Can you test with the replace fix available in the comment linked below ? it should fixes your problem as DOTALL does by removing the EOLs instead. Thanks.
https://github.com/home-assistant/home-assistant/issues/11142#issuecomment-364649288

**Comment by MatMaul** [comment]
@hartmannaf @OneGneissGuy tagging people for testing ;) Once I am confident that the replace fixes it for you too I am releasing 0.4.0.

**Comment by MatMaul** [comment]
Ok you can forget the test, I removed the regex and doing proper xml parsing instead. Please test with the 0.4.0 release to come and let me know.
